IssueService.java

```diff
@@ -0,0 +1,57 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class IssueService {
+
+	@Autowired
+	private IssueRepository issueRepository;
+
+	public Issue update(Issue issue) throws InvalidIssueStatusException {
+
+		// TODO: Validate the issue.status first;
+		// Status can be updated from CREATED to IN_PROGRESS to COMPLETED.
+		if(!isValidStatus(issue)){
+			throw new InvalidIssueStatusException("Issue status is invalid!");
+		}
+		
+		// TODO: Save the issue using issueRepository
+		return issueRepository.updateIssue(issue);
+	}
+	
+	/**
+	 * Checks if issue status is valid to update
+	 * 
+	 * @param issue
+	 * @return
+	 */
+	private boolean isValidStatus(Issue issue) {
+		try {
+			if (issue == null
+					|| issue.getStatus() == null
+					|| (IssueStatus.valueOf(issue.getStatus().toUpperCase())
+							- IssueStatus.valueOf(issueRepository.getCurrentStatus(issue).toUpperCase()) < 0)) {
```

Comment:
I don't think issueRepository will have getCurrentStatus(issue) method. Instead it generally has get(issueId) where you can get current issue and retrieve the status.

Reply:
written based on assumption that IssueRepository will take Issue object as input and provide the status, we can update this based on IssueRepository functionality.